sled-agent/config-reconciler/src/dump_setup.rs

```diff
@@ -1101,13 +1101,19 @@ impl DumpSetupWorker {
         let oxz_zones = self.zone_invoker.get_zones().await?;
         for zone in oxz_zones {
             let logdir = if zone.global() {
-                PathBuf::from("/var/svc/log")
+                zone.path.join("var/svc/log")
             } else {
                 zone.path().join("root/var/svc/log")
             };
             let zone_name = zone.name();
             self.archive_logs_from_zone_path(
-                debug_dir, logdir, zone_name, false,
+                debug_dir, logdir, "*.log", zone_name, false,
+            )
+            .await?;
+
+            let adm_logdir = zone.path.join("var/adm");
```

Comment:
This seems not quite right for non-global zones.

If I'm understanding right:

- for the global zone, `zone.path()` is `/`.  This is the zone's path and _also_ the zone's root filesystem path.
- for the non-global zone, `zone.path()` is something else.  This is the zone's path, but it's not the root filesystem.  That's under `root`.

I'd suggest adding up at L1102 or so:

```rust
let zone_root = if zone.global() { zone.path().to_owned() } else { zone.path().join("root"); };
```

Then you can just have:

```rust
let logdir = zone_root.join("var/svc/log");
```

for both cases.  And then here, you can use `zone_root.join("/var/adm")`.